Consul.Test/BaseFixture.cs

```diff
@@ -87,9 +91,9 @@ static BaseFixture()
             // XUnit sets the initial number of worker threads to the number of CPU cores.
             // Unfortunately, when the initial limit for the ThreadPool is too low, it introduces a risk of a deadlock-like behavior and the tests are timing out.
             ThreadPool.GetMinThreads(out var workerThreads, out var completionPortThreads);
-            if (workerThreads < 4)
+            if (workerThreads < 6)
```

Comment:
Occasionally, I've seen some semaphore-related tests fail. From logs, I concluded that it was caused by the starvation of a thread pool. I think that we should bump the available threads in the thread pool even more.

Reply:
Ok.. so why 6 and not more?

Reply:
I don't have a good answer here. I want to keep the number reasonably low, but the exact number is an arbitrary decision. Let's give it a try, if tests still fail then we can bump the number even more.